components/coding/profileV2/ProfileHeaderComponent.tsx

```diff
@@ -0,0 +1,124 @@
+import { useQuery } from "@apollo/client";
+import { format } from "date-fns";
+import React, { useState } from "react";
+import { FETCH_TOTAL_USER_BADGES_COUNT } from "../../../graphql/fetchTotalUserBadgesCount";
+import {
+  FetchUserBadgesCountResponse,
+  FETCH_USER_BADGES_COUNT,
+} from "../../../graphql/fetchUserBadgesCount";
+import {
+  UserProfileData,
+  FetchUserProfileDataResponse,
+  FETCH_USER_PROFILE_DATA,
+} from "../../../graphql/fetchUserProfile";
+import BadgesDisplayedComponent from "./BadgesDisplayedComponent";
+
+import JoinedDateComponent from "./JoinedDateComponent";
+
+export type UserProfileSectionProps = {
+  user: User;
+};
+
+export type User = {
```

Comment:
Nice job mapping this out this type! How about we move this type to the file that contains FETCH_USER_PROFILE_DATA